test/gtest/uct/ib/test_ib.cc

```diff
@@ -67,183 +166,102 @@ void test_uct_ib::init() {
-
-    my_desc->length = data_length;
-    if (*test_ib_hdr == 0xbeef) {
-        memcpy(my_desc + 1, actual_data , data_length);
-    }
-    ++test_uct_ib::m_ib_am_handler_counter;
-    return UCS_OK;
+void test_uct_ib_with_specific_port::cleanup() {
+    close_device();
+    test_uct_ib::cleanup();
 }
 
-bool test_uct_ib::test_eth_port() {
-    bool have_valid_gid_idx = false;
-
-    if (!IBV_PORT_IS_LINK_LAYER_ETHERNET(&m_port_attr)) {
-        return have_valid_gid_idx;
-    }
-   
-#if HAVE_DECL_IBV_LINK_LAYER_ETHERNET
-    union ibv_gid gid;
-    uct_ib_md_config_t *md_config = ucs_derived_of(m_md_config, uct_ib_md_config_t);
-    uct_md_h uct_md;
-    uct_ib_md_t *ib_md;
-    ucs_status_t status;
-    uint8_t gid_index;
-
-    status = uct_ib_md_open(ibv_get_device_name(m_ibctx->device), m_md_config,
-                            &uct_md);
-    ASSERT_UCS_OK(status);
-
-    ib_md = ucs_derived_of(uct_md, uct_ib_md_t);
-    status = uct_ib_device_select_gid_index(&ib_md->dev, m_port,
-                                            md_config->ext.gid_index,
-                                            &gid_index);
-    ASSERT_UCS_OK(status);
-
-    /* check the gid index */
-    if (ibv_query_gid(m_ibctx, m_port, gid_index, &gid) != 0) {
-        UCS_TEST_ABORT("Failed to query gid (index=" << gid_index << ")");
-    }
-    if (uct_ib_device_is_gid_raw_empty(gid.raw)) {
-        have_valid_gid_idx = false;
-    } else {
-        have_valid_gid_idx = true;
+class test_uct_ib_lmc : public test_uct_ib_with_specific_port {
+public:
+    void check_port_attr() {
+        /* check if a non zero lmc is set on the port */
+        if (!m_port_attr.lmc) {
+            UCS_TEST_SKIP_R("lmc is set to zero on an IB port");
+        }
     }
+};
 
-    uct_ib_md_close(uct_md);
-#endif
-
-    return have_valid_gid_idx;
-}
-
-bool test_uct_ib::lmc_find() {
-    return (m_port_attr.lmc > 0);
+UCS_TEST_P(test_uct_ib_lmc, non_default_lmc, "IB_LID_PATH_BITS=1") {
+    send_recv_short();
 }
 
-void test_uct_ib::test_address_pack(uint64_t subnet_prefix) {
-    uct_ib_iface_t *iface = ucs_derived_of(m_e1->iface(), uct_ib_iface_t);
-    static const uint16_t lid_in = 0x1ee7;
-    union ibv_gid gid_in, gid_out;
-    uct_ib_address_t *ib_addr;
-    uint16_t lid_out;
-
-    ib_addr = (uct_ib_address_t*)malloc(uct_ib_address_size(iface));
-
-    gid_in.global.subnet_prefix = subnet_prefix;
-    gid_in.global.interface_id  = 0xdeadbeef;
-    uct_ib_address_pack(iface, &gid_in, lid_in, ib_addr);
-
-    uct_ib_address_unpack(ib_addr, &lid_out, &gid_out);
-
-    if (uct_ib_iface_is_roce(iface)) {
-        EXPECT_TRUE(iface->is_global_addr);
-    } else {
-        EXPECT_EQ(lid_in, lid_out);
-    }
+UCT_INSTANTIATE_IB_TEST_CASE(test_uct_ib_lmc);
 
-    if (iface->is_global_addr) {
-        EXPECT_EQ(gid_in.global.subnet_prefix, gid_out.global.subnet_prefix);
-        EXPECT_EQ(gid_in.global.interface_id,  gid_out.global.interface_id);
+#if HAVE_DECL_IBV_LINK_LAYER_ETHERNET
```

Comment:
why do we need to put all test under this ifdef?
isn't using IBV_PORT_IS_LINK_LAYER_ETHERNET macro enough?